### PR TITLE
Baris/refactor beans

### PR DESCRIPTION
**Pull request by bbayazit16**
Refactored to simplify property listener logic.

### CODE REVIEW

**Review by HACKERALERT — APPROVED** [review]

File: src/main/java/views/TestView.java [hunk 1]
```diff
@@ -1,40 +1,54 @@
 package views;
 
-import interface_adapter.NavigableUiPanel;
+import service.ViewManagerModel;
 import views.components.DoorkeyButton;
 
 import javax.swing.*;
```
Comment (9naama) [inline src/main/java/views/TestView.java:6]: Fix * input check style error

File: src/main/java/views/TestView.java [hunk 2]
```diff
@@ -1,40 +1,54 @@
...
 import java.awt.*;
+import java.awt.event.ActionEvent;
```
Comment (9naama) [inline src/main/java/views/TestView.java:8]: adjust order as necessary to satisfy check style

File: src/main/java/views/TestViewModel.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package views;
+
+import interface_adapter.ViewModel;
+
+public class TestViewModel extends ViewModel<TestViewState> {
```
Comment (9naama) [inline src/main/java/views/TestViewModel.java:5]: fix "type Javadoc comment is missing null tag."

File: src/main/java/views/TestViewState.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
+package views;
```
Comment (9naama) [inline src/main/java/views/TestViewState.java:1]: file needs to end with new line

File: src/main/java/views/TestViewState.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+public class TestViewState {
```
Comment (9naama) [inline src/main/java/views/TestViewState.java:3]: address missing null tag

**Review by 9naama — CHANGES_REQUESTED** [review]
Check style error fixes left in comments, please address otherwise LGTM

**Review by 9naama — APPROVED** [review]

**Pull request merged**